plugin-mongodb/src/lib.rs

```diff
@@ -0,0 +1,179 @@
+        if measurements.is_empty() {
+            log::warn!("MongoDb recieved an empty MeasurementBuffer");
+            return Ok(());
+        }
+
+        // Build the data to send to MongoDb
+        let mut docs = vec![];
+        for m in measurements {
+            let mut doc = doc! {};
+            let metric = ctx.metrics.by_id(&m.metric).unwrap();
+            doc.insert("measurement", &metric.name);
+
+            // Resources and consumers are translated as fields
+            doc.insert("resource_kind", m.resource.kind());
+            doc.insert("resource_id", m.resource.id_string().unwrap_or_default());
+            doc.insert("resource_consumer_kind", m.consumer.kind());
+            doc.insert("resource_consumer_id", m.consumer.id_string().unwrap_or_default());
+
+            // Alumer attributes are translated to fields
+            // Some field keys are reserved by Alumet and will trigger a renaming
+            let attributes = m.attributes();
+            const RESERVED_FIELDS: [&str; 6] = [
+                "measurement",
+                "resource_kind",
+                "resource_id",
+                "resource_consumer_kind",
+                "resource_consumer_id",
+                "value",
+            ];
+
+            for (field_key, field_value) in attributes {
+                let field_key = if RESERVED_FIELDS.contains(&field_key) {
+                    format!("{}_field", field_key)
+                } else {
+                    String::from(field_key)
+                };
+                match field_value {
+                    AttributeValue::F64(v) => {
+                        doc.insert(field_key, v.to_string());
+                    }
+                    AttributeValue::U64(v) => {
+                        doc.insert(field_key, format!("{v}u"));
+                    }
+                    AttributeValue::Bool(v) => {
+                        doc.insert(field_key, if *v { "T" } else { "F" });
+                    }
+                    AttributeValue::Str(v) => {
+                        doc.insert(field_key, v);
+                    }
+                    AttributeValue::String(v) => {
+                        doc.insert(field_key, v);
+                    }
+                }
+            }
+
+            // Append alumet value
+            match m.value {
+                WrappedMeasurementValue::F64(v) => {
+                    doc.insert("value", v.to_string());
+                }
+                WrappedMeasurementValue::U64(v) => {
+                    doc.insert("value", format!("{v}u"));
+                }
+            }
+
+            // Add the timestamp
+            doc.insert("timestamp", convert_timestamp(m.timestamp));
+
+            docs.push(doc);
+        }
+
+        // Send the data to MongoDb
+        let collection = self.client.database(&self.database).collection(&self.collection);
+        let rt = tokio::runtime::Runtime::new().unwrap();
```

Comment:
You should not create a new normal runtime on each `write`, because `write` is already called in a runtime by Alumet. Either you use the current-thread runtime, or you turn your output into an "async" output (See `AlumetPluginStart::add_async_output_builder`).